controllers/resources.go

```diff
@@ -816,3 +834,59 @@ func getAggregateToViewPolicyRules() []rbacv1.PolicyRule {
 		},
 	}
 }
+
+func (r *RolloutManagerReconciler) createServiceMonitorIfAbsent(namespace string, rolloutManager *rolloutsmanagerv1alpha1.RolloutManager, name, serviceMonitorLabel string) error {
+	existingServiceMonitor := &monitoringv1.ServiceMonitor{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, existingServiceMonitor)
+	if err == nil {
+		log.Info("A ServiceMonitor instance already exists",
+			"Namespace", existingServiceMonitor.Namespace, "Name", existingServiceMonitor.Name)
+		return nil
```

Comment:
As above, if the ServiceMonitor does exist, we should verify it's content, and fix it if it's out of sync.